File: Bit-Manipulation/single-number-2/solution.py
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        score = {}
        for num in nums:
            score[num] = score[num] + 1 if num in score else 1
        for (key,value) in score.items():
            if value < 3:
                return key

File: Bit-Manipulation/single-number-2/solution.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var singleNumber = function (nums) {
  const map = new Map();
  for (let num of nums) {
    const count = map.get(num) ? map.get(num) + 1 : 1;
    map.set(num, count);
  }
  for (let [key, value] of map.entries()) {
    if (value < 3) return key;
  }
};
